server/app/views/admin/programs/ProgramTranslationView.java

```diff
@@ -158,6 +161,64 @@ private ImmutableList<DomContent> formFields(
                           .asAnchorText()),
               fieldsBuilder.build()));
     }
+
+    // Add slim alert with warning that translations aren't visible yet.
+    result.add(
+        div()
+            .withClasses("usa-alert", "usa-alert--info", "usa-alert--slim")
+            .with(
+                div()
+                    .withClass("usa-alert__body")
+                    .with(
+                        p().withClass("usa-alert__text")
+                            .withText(
+                                "Translations entered below will be visible at a future launch"
+                                    + " date."))));
+
+    // Add fields for Screen names and descriptions
+    for (int i = 0; i < updateData.screens().size(); i++) {
+      LocalizationUpdate.ScreenUpdate screenUpdateData = updateData.screens().get(i);
+
+      BlockDefinition block =
+          program.blockDefinitions().stream()
```

Comment:
**suggestion:** Can we store `program.blockDefinitions()` or `program.blockDefinitions().stream()` in a variable above the for loop, so that we don't have to make the call to get the block definitions for each screen?

Reply:
done